plugins/python/python_plugin.c

```diff
@@ -1297,29 +1288,49 @@ void uwsgi_python_init_apps() {
 
 }
 
-void uwsgi_python_master_fixup(int step) {
 
-	static int master_fixed = 0;
-	static int worker_fixed = 0;
+void uwsgi_python_pre_uwsgi_fork() {
+	if (!uwsgi.master_process) return;
+
+	if (uwsgi.has_threads) {
+		// Since we are not calling cPython's fork, we need to avoid forking
+		// while the gil and import lock are held by a thread other than the
+		// main thread.
```

Comment:
```suggestion
		// while the gil and import lock are held by by a Python thread
```

? wondering how we might be able to clarify what "main thread" means here


```suggestion
		// while the gil and import lock are held by a thread other than the
		// uWSGI main thread.
```

?